std/container/array.d

```diff
@@ -539,11 +525,13 @@ Complexity: $(BIGOH 1)
     }
 
 /**
-Ensures sufficient capacity to accommodate $(D e) elements.
+Ensures sufficient capacity to accommodate $(D e) _elements.
+If $(D e < capacity), this method will not decrease the capacity.
 
 Postcondition: $(D capacity >= e)
 
-Complexity: $(BIGOH 1)
+Complexity: $(BIGOH e) if $(D e > capacity),
```

Comment:
> reserve just makes sure there is enough capacity but the internal algorithm of Array, should, of course, increase capacity using some proportional progression so as to minimize reallocations.

I agree that `insertBack` should do that...

> One more reason to avoid std.container for the time being ...

and of course it does so and we can without a doubt say that it is amortized constant, as @klickverbot said, no worries.

I'm not so sure about `reserve` - personally I'd like to have a possibility to allocate exactly as much memory as I need. Taking look at C++ [vector.reserve](http://en.cppreference.com/w/cpp/container/vector/reserve) docs, they warn about inapropriate use of `reserve` and declare complexity as `O(size)` (although they don't use `realloc` as far as I know, but always allocate new chunk of memory if capacity grows).